src/test/java/org/roaringbitmap/TestRunContainer.java

```diff
@@ -2969,5 +2988,36 @@ public void lazyOR() {
     assertEquals(1<<16, rc.getCardinality());
   }
 
+  @Test
+  public void testLazyORFull() {
+    Container rc = Container.rangeOfOnes(0, 1 << 15);
+    BitmapContainer bc2 = new BitmapContainer(3210, 1 << 16);
+    Container rbc = rc.lazyOR(bc2);
+    assertEquals(-1, rbc.getCardinality());
+    Container repaired = rbc.repairAfterLazy();
+    assertEquals(1 << 16, repaired.getCardinality());
+    assertThat(repaired, instanceOf(RunContainer.class));
+  }
+
+  @Test
+  public void testLazyORFull2() {
+    Container rc = Container.rangeOfOnes(0, 1 << 15);
+    ArrayContainer ac = new ArrayContainer((1 << 15) - 200, 1 << 16);
+    Container rbc = rc.lazyOR(ac);
```

Comment:
There is an optimization missing here. Have a look at rbc... Given the good job you've done so far, it will be easy for you to understand what I mean :wink: